src/main/java/org/sudoku/Sudoku.java

```diff
@@ -1,143 +1,150 @@
-          if(!clonedIntersectionNumbers.isEmpty()){
-            removeNumbersFromCells(changedCells, substractionSquareFromLaneCells, clonedIntersectionNumbers);
-          }
-          intersectionNumbers.removeAll(substractionSquareFromLaneNumbers);
-          if(!intersectionNumbers.isEmpty()) {
-            removeNumbersFromCells(changedCells, substractionLaneFromSquareCells, intersectionNumbers);
-          }
+
+    public void recalcCell(Cell a){
+        a.possibleValues.clear();
+        for(int i = 1; i < 10; i++){
+            if(fitTest(a, i)){
+                a.possibleValues.add(i);
+            }
         }
-      }
-      reorderCells(changedCells);
-      result = !changedCells.isEmpty();
     }
-  }
 
-  private void checkCellsWithSinglePossibleNumber() {
-    Iterator<Cell> cellIterator = cells.stream().filter(cell -> cell.possibleNumbersSize() == 1).iterator();
+    public boolean fitTest(Cell a, int n){
+        List<Cell> horiz = cells.stream().filter(c->c.getY()==a.getY()).collect(Collectors.toList());
+        List<Cell> vert = cells.stream().filter(c->c.getX()==a.getX()).collect(Collectors.toList());
+        List<Cell> sq = cells.stream().filter(c->c.getSquare()==a.getSquare()).collect(Collectors.toList());
+        for (Cell ce : horiz) {
+            if(ce.getVal() == n){
+                return false;
+            }
+        }
+        for (Cell ce : vert) {
+            if(ce.getVal() == n){
+                return false;
+            }
+        }
 
-    while (cellIterator.hasNext()) {
-      Cell firstCell = cellIterator.next();
-      Byte value = firstCell.getPossibleNumbers().stream().findFirst().get();
-      Set<Cell> updatedCells = firstCell.setNumberValue(value);
-      reorderCells(updatedCells);
-      cellIterator = cells.stream().filter(cell -> cell.possibleNumbersSize() == 1).iterator();
+        for (Cell ce : sq) {
+            if(ce.getVal() == n){
+                return false;
+            }
+        }
+        return true;
     }
-  }
-
-  private void removeNumbersFromCells(Set<Cell> changedCellsCollector, Set<Cell> cells, Set<Byte> numbers) {
-    for(Cell cell: cells) {
-      for (Byte number : numbers) {
-        if (cell.removePossibleNumber(number))
-          changedCellsCollector.add(cell);
-      }
+
+    public List<Cell> solveLVL1(){
+        this.cells.stream().filter(cell -> cell.getVal()==0).forEach(cel-> this.recalcCell(cel));
+
+        List<Cell> easyCells = cells.stream().filter(c->c.possibleValues.size() == 1).collect(Collectors.toList());
+        for (Cell cel : easyCells) {
+            cel.setValue(cel.possibleValues.get(0));
+        }
+        return easyCells;
     }
-  }
-
-  public void print() {
-    String sep = getSeparatorLane();
-    System.out.println(sep);
-    for(Lane lane: hlanes) {
-      System.out.println(lane.toString());
-      System.out.println(sep);
+
+    public boolean isLast(){
+        for (Cell cell : this.cells) {
```

Comment:
I guess this is the cell that was not tested yet - possibly more efficient to store index in the list vs iterating each time